sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -611,7 +611,8 @@ public String getEntityPath() {
             withAutoComplete = withAutoLockRenewal;
         }
 
-        return withAutoComplete;
+        return withAutoComplete
+            .onErrorMap(throwable -> mapError(throwable, ServiceBusErrorSource.RECEIVE));
```

Comment:
* Should map AutoComplete and AutoLockRenew Mono<> with the error mapping.
* This is going to mask an autocomplete/Renew exception. iirc, we pass it downstream. it would not be a receive error, it would be a complete error.
This should be different in that we check the exception type, if it's ServiceBusErrorSource already and if so, not map it.